Reregistration GR-PEACH as mbed support board

TomoYamanaka opened the pull request:
## Description
By #5342, since Cortex-A was supported wtih updating CMSIS/RTX, I request that **GR-PEACH** is registed as mbed board again.

I've closed #5530, because I removed my commits accidently.  So, I made this PR again.
<br>

The main changes is the below.
- Changes to PEACH code (4 commits)
- Changes to Mbed OS including to reverting "disabling the Cortex-A" (5 commits)
- Cherry-picked changes from CMSIS/RTX repo (3 commits)
- New changes to CMSIS/RTX (3 commits)

## Request
- I strongly desire to be included in the shortest release package after completing the merge of this PR.

##
@toyowata 
Please help me to go ahead with the above.

0xc0170 commented:
I looked at the changes (changes to target code looks fine to me). Got question regarding changes to CMSIS/RTX. 

The last 3 commits - are they going also upstream (are there pull requests or issues reported) ?

3 commits that are being cherry-picked from CMSIS repository, they are already released (what version), or develop branch ? To be aware, what update do we need to get these in.

bulislaw commented:
@jonatanantoni could you review this 3 commits, as if they would be submitted to CMSIS_5 repo:
https://github.com/ARMmbed/mbed-os/pull/5628/commits/c6fa97c45ac6704a68a8ce1a92684d5dc098926a
https://github.com/ARMmbed/mbed-os/pull/5628/commits/c91b39a5ee0ade182f0087017c3bddbcab02cb70
https://github.com/ARMmbed/mbed-os/pull/5628/commits/aaad6c170b3b90b3d04f0cfc92a6c0f459278e7a

bulislaw commented on cmsis/TARGET_CORTEX_A/core_ca.h:
```diff
@@ -1788,6 +1788,20 @@ typedef struct RegionStruct {
                                    region.sh_t = NON_SHARED; \
                                    MMU_GetSectionDescriptor(&descriptor_l1, region);
 
+#define section_normal_nc(descriptor_l1, region)     region.rg_t = SECTION; \
```
Just looking at the different macros this one is missing documentation

TomoYamanaka commented:
@bulislaw @JonatanAntoni 
For merge the changes([0421868](https://github.com/ARMmbed/mbed-os/pull/5628/commits/0421868ee7b385659deb647bffb8ff400b557bf2), [fefec98](https://github.com/ARMmbed/mbed-os/pull/5628/commits/fefec9845dd0d4344456d501083d9208ada7386b), [e2acfc1](https://github.com/ARMmbed/mbed-os/pull/5628/commits/e2acfc1246b7a28794242baab20e8d2338d3d4f9)) of CMSIS/RTX, I has been promoted discussion by giving three PR to CMSIS_5 repo.
Currently, one of three([#286](https://github.com/ARM-software/CMSIS_5/pull/286)) was merged in CMSIS repo, but rest requests([#287](https://github.com/ARM-software/CMSIS_5/pull/287), [#288](https://github.com/ARM-software/CMSIS_5/pull/288)) do not be still merged and discussion is continuing. It seems that the cause and my changing contents obtained a certain understanding, now discussion focus on permanent fix. I think that permanent fix is needed, however, I strongly desire to merge this PR to Mbed OS 5.7 that will mention "cortex-A suppport" as official. 

Therefore, Could you please merge my changes of CMSIS/RTX as tentative?
Of course, if permanent fix was completed in CMSIS_5 repo, I will cherry-pick them.

@toyowata
Please help me to go ahead with the above.

0xc0170 commented:
> For merge the changes(0421868, fefec98, e2acfc1) of CMSIS/RTX, I has been promoted discussion by giving three PR to CMSIS_5 repo.

Thank you 👍 

> Currently, one of three(#286) was merged in CMSIS repo, but rest requests(#287, #288) do not be still merged and discussion is continuing. It seems that the cause and my changing contents obtained a certain understanding, now discussion focus on permanent fix. I think that permanent fix is needed, however, I strongly desire to merge this PR to Mbed OS 5.7 that will mention "cortex-A suppport" as official.

Based on the review for those opened 2 PR, and discussion with @JonatanAntoni. There's only one left, 288 was fixed  , so that could be done here the same (cosmetic change there the functionality the same). Thus only 287 needs further validation. Please help there to validate.

Thus only one remaining, that is not blocking this PR to my understanding.

What I would like  to have one documentation improvement, `New changes to CMSIS/RTX (No 1 of total 3 commits) ` The headline for each commit should provide meaningful information to a reader within 50 characters. If you could rework the commit messages it would be helpful. In case of `New changes to CMSIS/RTX (No 1 of total 3 commits) ` this could be `CMSIS: add section_normal_nc impl` or similar then the paragraph below is good as you have it 👍

TomoYamanaka commented:
@0xc0170 
Thank you for your comments.
I rebased those commits in the form of changing commit message. 

> Based on the review for those opened 2 PR, and discussion with @JonatanAntoni. There's only one  left, 288 was fixed , so that could be done here the same (cosmetic change there the functionality the same). Thus only 287 needs further validation. Please help there to validate.

286 and 288 were fixed in CMSIS_5 repo so that I cherry-picked those two commits([5f850a2](https://github.com/ARMmbed/mbed-os/pull/5628/commits/5f850a2cab3fdf74c055695cbdf9043f0cf810d1) , [839e4a5](https://github.com/ARMmbed/mbed-os/pull/5628/commits/839e4a55e391b9c5409be7f2bed2ba5c4ba4dfd3)).
Also since only 287 needs further validation, I didn't cherry-pick, and changed only the commit message.

Please let me know if there is anything you are unclear.

toyowata commented:
@0xc0170 Any additional step required for this PR?

0xc0170 commented:
@TomoYamanaka If you can please rebase, we will trigger tests

TomoYamanaka commented:
@0xc0170 @toyowata 

I rebased those commits.

TomoYamanaka commented:
How is going? Is there anything else to do?

0xc0170 commented:
/morph build

0xc0170 commented:
It seems our CI got a license problem, will investigate

cc @studavekar @kegilbert

TomoYamanaka commented:
@0xc0170 
I changed the linker script files to implement the dynamic HEAP the same as Cortex-M targets.
Since GR-PEACH's HEAP was a fixed area, I changed the label name(ZI_DATA to RW_IRAM1) and replaced the allocation of STACK/HEAP.

Do you know what I mean?
If there is no problem in this description, I will update the commit message soon.

TomoYamanaka commented:
@0xc0170 
Thank you. I updated the commit msg.

bulislaw commented:
@TomoYamanaka we still have two commits touching CMSIS/RTX code:
https://github.com/ARMmbed/mbed-os/pull/5628/commits/a888ca5b88014fe5b960f0c6a6872b32f5fa825a
https://github.com/ARMmbed/mbed-os/pull/5628/commits/4d85759248818873d80ef4aea74d589981a11815

Where are we with pushing them to the original CMSIS/RTX repo?

TomoYamanaka commented:
@bulislaw 

- https://github.com/ARMmbed/mbed-os/commit/a888ca5b88014fe5b960f0c6a6872b32f5fa825a
I cherry-picked [this commit](https://github.com/ARM-software/CMSIS_5/commit/a76e53cbaec97f641357051394e118fd976832d3) from CMSIS_5 repo.


- https://github.com/ARMmbed/mbed-os/commit/4d85759248818873d80ef4aea74d589981a11815
I don't cherry-pick this commit from CMSIS_5 repo. 
I aim to merge this PR to Mbed OS 5.7, so I discussed with Arm Mbed team member on the thread and agreed to merge this commit as a tentative.

JonatanAntoni commented:
@TomoYamanaka,

4d85759 should have been superseded by 17b53ef2, bd0db8d and d43938f.

Cheers,
Jonatan

TomoYamanaka commented:
@JonatanAntoni 

I rebased the commit `4d85759` by cherry-pick those commits([17b53ef2](https://github.com/ARM-software/CMSIS_5/commit/17b53ef2a297ef73d0fb0d720cf2a3458d6127e2)  , [d43938f](https://github.com/ARM-software/CMSIS_5/commit/d43938fac6e0fa1b5682c2ca5704277b6dc34963)).

bulislaw commented:
@TomoYamanaka 

> 4d85759
I don't cherry-pick this commit from CMSIS_5 repo.
I aim to merge this PR to Mbed OS 5.7, so I discussed with Arm Mbed team member on the thread and agreed with them to merge this commit as a tentative.

Could you point me to the comment?

bulislaw commented:
I mean this commit now https://github.com/ARMmbed/mbed-os/pull/5628/commits/a888ca5b88014fe5b960f0c6a6872b32f5fa825a

bulislaw commented:
Ah sorry! It seems all the commits are cherrypicked now. @0xc0170 are you ok with this PR now?

toyowata commented:
@0xc0170 @bulislaw can you please remove review flag and trigger CI test?

0xc0170 commented:
@TomoYamanaka Can you please rebase to fetch the fix for travis timeout from master, we will trigger CI tests

TomoYamanaka commented:
@0xc0170 
I rebased those commits. Is this fine with you?

bulislaw commented:
/morph build

bulislaw commented:
Arm compiler licence issues - retry.

/morph build

0xc0170 commented:
Restarted uvisor job (flash failure). mbed-os-generic-context - to be resolved, should not be here

0xc0170 commented:
@orenc17A Can you help resolving an error in uvisor CI?

orenc17A commented:
/morph uvisor-test

toyowata commented:
@0xc0170 Can you merge this please?

bulislaw commented:
@0xc0170 @theotherjimmy

adbridge commented:
This relies on previous changes targeted for 5.8 so should also go to that minor release.

0xc0170 commented:
> This relies on previous changes targeted for 5.8 so should also go to that minor release.
 
Please provide reference of offending dependencies.

adbridge commented:
Having done a re-run this is now patching across ok. Not sure why it was failing before unless it was conflicting with another PR which may have also been removed!
